### PR TITLE
Attach timer and message to activity boundaries

### LINKED ISSUE #31: I can attach timer/message catch event to activity boundaries
With version 0.4.0, it seems I can't drop a timer/message event directly on the boundary of an activity. It won't allow me dropping it on a task, and on a sub-process it just adds it as a child element.

The workaround is to drop a none event and switch its type, which is OK, but not great UX.

I think it'd be more intuitive to allow dropping the events directly on the boundary of activities.

### PR DESCRIPTION
**Pull request by dominikhrn**
__Which issue does this PR address?_

Closes #31

### CODE REVIEW

**Comment by dominikhrn** [comment]
@pinussilvestrus Does Anyone want to review that feature? Not sure who I should assign for that...

**Comment by dominikhrn** [comment]
Thanks for your feedback @pinussilvestrus ! I'll take a look at that 👍

File: client/src/app/tabs/zeebe/custom/CustomRules.js [hunk 1]
```diff
@@ -1,63 +1,143 @@
...
+function isBoundaryEvent(element) {
+  return !isLabel(element) && is(element, 'bpmn:BoundaryEvent');
 }
 
 /**
- * Specific rules for custom elements
+ * In Zeebe we treat IntermediateCatchEvents as boundary events too,
+ * this must be reflected in the rules.
  */
-export default class CustomRules extends RuleProvider {
```
[the remark this answers is not in the document]
Comment (dominikhrn) [reply client/src/app/tabs/zeebe/custom/CustomRules.js]: I'll change that, thanks for the hint! Was not my intention to do it that way.

**Comment by pinussilvestrus** [comment]
Rebased to `@master`.

**Comment by nikku** [comment]
Rebased on top of `master`.

File: client/src/app/tabs/bpmn/custom/CustomRules.js [hunk 2]
```diff
@@ -1,63 +1,143 @@
...
+      context.target,
+      null,
+      context.position);
 
-function hasOutgoings(element) {
-  return element.outgoing && element.outgoing.length > 0;
+  });
+};
+
+BpmnRules.prototype.canAttach = canAttach;
```
[the remark this answers is not in the document]
Comment (dominikhrn) [reply client/src/app/tabs/bpmn/custom/CustomRules.js]: Hi @nikku, the problem ist that the canAttach Method of the BpmnRules is used by other Modules. (BpmnSnapping)
I did try to totaly replace the BpmnRules by the CustomRules. That didn't work for some reason...

[the remark this answers is not in the document]
Comment (nikku) [reply client/src/app/tabs/bpmn/custom/CustomRules.js]: What did not work about replacing the rules completely? You'd need to simply override the definition of `bpmnRules` with the `CustomRules` component (as shown https://github.com/zeebe-io/zeebe-modeler/pull/44/files#diff-a9fcb3312ff27119bbbffd989a060690R7) and ensure that all BpmnRules related initializations happen by calling super() where appropriate.

[the remark this answers is not in the document]
Comment (dominikhrn) [reply client/src/app/tabs/bpmn/custom/CustomRules.js]: You are absolutely right. I did it in the es6 way and it worked like a charm. If you have any suggestions for improvement, just tell me.

File: client/src/app/tabs/zeebe/custom/CustomRules.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (dominikhrn) [reply client/src/app/tabs/zeebe/custom/CustomRules.js]: Changed it to es6.

File: client/src/app/tabs/bpmn/custom/CustomRules.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (pinussilvestrus) [reply client/src/app/tabs/bpmn/custom/CustomRules.js]: awesome! I'll add some test coverage for the CustomRules and then it should be ready to merge!

**Comment by pinussilvestrus** [comment]
@dominikhrn 
I just added some test coverage for `CustomRules` + make sure that the `CustomRules` really got triggered inside the `CreateZeebeBoundaryEventBehaviorSpec`. They seemed to be not, since I found out that just the originally `BpmnRules` were used inside the spec, which was caused just because the custom modules have to included right after modeling.

However, the PR should be ready for merge now. 🙂 Maybe @nikku can have a final look, otherwise just merge it.

**Comment by nikku** [comment]
Final changes applied on the PR:

* Re-created an updated screenshot via [screely](https://screely.com), the tool we use across all our projects
* Re-built the package-lock; to install things without modifying it do `npm ci` instead of `npm install`

**Pull request merged**